backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/material/cockpit/MD_Cockpit_DocumentDetail_StepDef.java

```diff
@@ -0,0 +1,210 @@
+		softly.assertThat(cockpitRecord).isNotNull();
+
+		final String orderLineIdentifier = DataTableUtil.extractStringForColumnName(tableRow, I_MD_Cockpit_DocumentDetail.COLUMNNAME_C_OrderLine_ID + "." + TABLECOLUMN_IDENTIFIER);
+		final I_C_OrderLine orderLineRecord = orderLineTable.get(orderLineIdentifier);
+		softly.assertThat(orderLineRecord).isNotNull();
+
+		final I_MD_Cockpit_DocumentDetail mdCockpitDocumentDetailRecord = getCockpitDocumentDetailByCockpitIdAndOrderLineId(timeoutSec,
+																															cockpitRecord.getMD_Cockpit_ID(),
+																															orderLineRecord.getC_OrderLine_ID());
+
+		final ItemProvider<I_MD_Cockpit_DocumentDetail> getValidCockpitDocumentDetail = () -> {
+			InterfaceWrapperHelper.refresh(mdCockpitDocumentDetailRecord);
+
+			return validateCockpitDocumentDetailRecord(tableRow, mdCockpitDocumentDetailRecord);
+		};
+
+		StepDefUtil.tryAndWaitForItem(timeoutSec, 500, getValidCockpitDocumentDetail, () -> logCurrentContext(cockpitRecord.getMD_Cockpit_ID(), orderLineRecord.getC_OrderLine_ID()));
+
+		softly.assertAll();
+	}
+
+	@NonNull
+	private ItemProvider.ProviderResult<I_MD_Cockpit_DocumentDetail> validateCockpitDocumentDetailRecord(
+			@NonNull final Map<String, String> tableRow,
+			@NonNull final I_MD_Cockpit_DocumentDetail cockpitDocumentDetailRecord)
+	{
+		final List<String> errorCollector = new ArrayList<>();
+
+		final String identifier = DataTableUtil.extractStringForColumnName(tableRow, I_MD_Cockpit_DocumentDetail.COLUMNNAME_MD_Cockpit_DocumentDetail_ID + "." + TABLECOLUMN_IDENTIFIER);
+
+		final BigDecimal qtyOrdered = DataTableUtil.extractBigDecimalOrNullForColumnName(tableRow, "OPT." + I_MD_Cockpit_DocumentDetail.COLUMNNAME_QtyOrdered);
+		if (qtyOrdered != null && !qtyOrdered.equals(cockpitDocumentDetailRecord.getQtyOrdered()))
+		{
+			errorCollector.add(MessageFormat.format("MD_Cockpit_DocumentDetail.Identifier={0}; Expecting QtyOrdered={1} but actual is {2}",
+													identifier, qtyOrdered, cockpitDocumentDetailRecord.getQtyOrdered()));
+		}
+
+		final BigDecimal qtyReserved = DataTableUtil.extractBigDecimalOrNullForColumnName(tableRow, "OPT." + I_MD_Cockpit_DocumentDetail.COLUMNNAME_QtyReserved);
+		if (qtyReserved != null && !qtyReserved.equals(cockpitDocumentDetailRecord.getQtyReserved()))
+		{
+			errorCollector.add(MessageFormat.format("MD_Cockpit_DocumentDetail.Identifier={0}; Expecting QtyReserved={1} but actual is {2}",
+													identifier, qtyReserved, cockpitDocumentDetailRecord.getQtyReserved()));
+		}
+
+		if (errorCollector.size() > 0)
+		{
+			final String errorMessages = String.join(" && \n", errorCollector);
+			return ItemProvider.ProviderResult.resultWasNotFound(errorMessages);
+		}
+
+		return ItemProvider.ProviderResult.resultWasFound(cockpitDocumentDetailRecord);
+	}
+
+	@NonNull
+	private I_MD_Cockpit_DocumentDetail getCockpitDocumentDetailByCockpitIdAndOrderLineId(
+			final int timeoutSec,
+			final int cockpitId,
```

Comment:
better use the RepoIdAwares rather than the integers (risk of mixing up paremeter ordering)